--- hostapd-2.4/src/ap/acs.c
+++ hostapd-2.4/src/ap/acs.c
@@ -661,7 +661,7 @@ acs_find_ideal_chan(struct hostapd_iface *iface)
 				   chan->chan);
 			continue;
 		}
-
+#if 0
 		/* 2.4 GHz has overlapping 20 MHz channels. Include adjacent
 		 * channel interference factor. */
 		if (is_24ghz_mode(iface->current_mode->mode)) {
@@ -716,7 +716,7 @@ acs_find_ideal_chan(struct hostapd_iface *iface)
 			tmp_bias.bias = ACS_24GHZ_PREFER_1_6_11;
 			bias = &tmp_bias;
 		}
-
+#endif
 		if (bias) {
 			factor *= bias->bias;
 			wpa_printf(MSG_DEBUG,
-- 
1.9.1

